src/ghga_connector/core/exceptions.py

```diff
@@ -180,184 +223,158 @@ def __init__(self, *, file_id: str):
+    """Raised when an issue is encountered during the initialization of a multipart upload"""
 
 
-class InvalidWorkPackageToken(RuntimeError):
-    """Thrown, when the work package string pasted by the user could not be parsed"""
+class UnauthorizedAPICallError(RuntimeError):
+    """Thrown when a 403 is returned from a call requiring a work order token for authorization"""
 
-    def __init__(self, *, tries: int):
-        message = f"Parsing of the work package string failed ({tries}) times."
+    def __init__(self, *, url: str, cause: str):
+        message = f"Could not authorize call to '{url}': {cause}"
         super().__init__(message)
 
 
-class NoWorkPackageAccessError(RuntimeError):
+class UnexpectedRetryResponseError(RuntimeError):
     """
-    Thrown when the given auth token does not provide access for
-    a specific work package id (response code 403)
+    Thrown when an unexpected RetryResponse was received while requesting URLs for a
+    staged download
     """
 
-    def __init__(self, *, work_package_id: str):
+    def __init__(self):
         message = (
-            "This auth token is not valid "
-            f"for the work package with the id '{work_package_id}'."
+            "An unexpected RetryResponse was received while requesting URLs for a"
+            + " staged download"
         )
         super().__init__(message)
 
 
-class InvalidWPSResponseError(RuntimeError):
-    """
-    Thrown when communication with the Work Package Service returns an unexpected response.
-    This should be used instead of BadResponseError when handling WPS results to differentiate.
-    """
+class UploadIdUnsetError(RuntimeError):
+    """Thrown when the upload ID was not set for operations requiring a valid upload ID."""
 
-    def __init__(self, *, url: str, response_code: int):
-        self.response_code = response_code
-        message = (
-            f"The request to the WPS at '{url}' failed with an unexpected response code "
-            + f"of {response_code}."
-        )
+    def __init__(self):
+        message = "Upload ID is not set, upload was not initialized correctly."
         super().__init__(message)
 
 
-class UnauthorizedAPICallError(RuntimeError):
-    """Thrown when a 403 is returned from a call requiring a work order token for authorization"""
+class UploadNotRegisteredError(RuntimeError):
+    """Thrown, when a request for a multipart upload returns a 404 error."""
 
-    def __init__(self, *, url: str, cause: str):
-        message = f"Could not authorize call to '{url}': {cause}"
+    def __init__(self, *, upload_id: str):
+        message = (
+            f"The request for the upload with the id '{upload_id}' failed, "
+            "because this upload does not exist."
+        )
         super().__init__(message)
 
 
-class EncryptedSizeMismatch(RuntimeError):
-    """Thrown when the actual encrypted size of a file does not match the computed one"""
+class UserHasNoFileAccessError(RuntimeError):
```

Comment:
Maybe easier to read without the "UserHasNo" prefix.

Also I wonder whether this should be merged with UserHasNoUploadAccessError because both are only used when uploading files.

Reply:
I agree on the first part, but would keep them separate for now.

There is a distinction being made based on the endpoints used and if the upload is already ongoing or not.
Maybe the `UserHasNoFileAccessError` needs a better name in general.

Reply:
Maybe you can find better names that express the difference then. The current names sound like one is for upload and the other one more general, but it's also only for upload.

Reply:
For now I removed the `UserHas` prefix and opted to make things a bit more clear in the docstring, as I could not think of a better fitting name